bioconda_utils/lint_functions.py

```diff
@@ -301,10 +301,11 @@ def pin(recipe, meta, df):
                 section = "run"
             elif in_requirements and line.strip().startswith("build:"):
                 section = "build"
-            elif not line.startswith(" ") and not line.startswith("#"):
+            elif not line.startswith(" ") and not line.startswith("#") and line.strip():
```

Comment:
To avoid redundancy and be more conservative, can we use:
```diff
+            elif line and not line.startswith(" ") and not line.startswith("#"):
```
? Note that only the [space character is considered as indenation](http://www.yaml.org/spec/1.2/spec.html#id2777534) so we should use `line.lstrip(" ")`. But since we already test for `line.startswith(" ")`, the only missing case is the empty string.

Reply:
Ah, I just now noticed the `line.strip()`s in the lines above. Let's just avoid that redundancy altogether and do a
```python
dedented_line = line.lstrip(" ")
```
and use `dedented_line` everywhere `line.strip()` is used.